### PR TITLE
Fix/dsp reg poll delay

### PR DESCRIPTION
**Pull request by fredoh9**
this is a fix for linux issue #517

### CODE REVIEW

**Comment by ranj063** [comment]
@fredoh9 congratulations on your first PR!

File: sound/soc/sof/intel/hda.h [hunk 1]
```diff
@@ -201,6 +201,7 @@
 #define HDA_ROM_INIT_TIMEOUT		70
 #define HDA_DSP_CTRL_RESET_TIMEOUT		100
 #define HDA_DSP_WAIT_TIMEOUT		500	/* 500 msec */
+#define HDA_DSP_REG_POLL_DELAY		500	/* 0.5 msec */
```
Comment (ranj063) [inline sound/soc/sof/intel/hda.h]: @fredoh9 I think we should call it initial delay and maybe add _US to the name to name it clear it is in us

File: sound/soc/sof/ops.c [hunk 2]
```diff
@@ -163,7 +165,7 @@ int snd_sof_dsp_register_poll(struct snd_sof_dev *sdev, u32 bar, u32 offset,
 
 		/* Phase 2 after 5ms(500us * 10) */
 		if (++k > 10)
-			s = 5000;
+			s = delay_us*10;
```
Comment (ranj063) [inline sound/soc/sof/ops.c]: checkpatch issue here. needs spaces around the *

File: sound/soc/sof/ops.h [hunk 3]
```diff
@@ -375,7 +375,7 @@ int snd_sof_pci_update_bits(struct snd_sof_dev *sdev, u32 offset,
 			    u32 mask, u32 value);
 
 int snd_sof_dsp_register_poll(struct snd_sof_dev *sdev, u32 bar, u32 offset,
-			      u32 mask, u32 target, u32 timeout);
+			      u32 mask, u32 target, u32 timeout, u32 delay);
```
Comment (ranj063) [inline sound/soc/sof/ops.h]: we should rename it to init_delay or something more indicative of what it is

File: sound/soc/sof/intel/hda.h (hunk 1 above)
Comment (ranj063) [reply sound/soc/sof/intel/hda.h]: I think resolution a more appropriate word than delay but it is too long. so maybe shorten it?

**Comment by ranj063** [comment]
@fredoh9 congratulations on your first PR!

**Comment by ranj063** [comment]
@fredoh9 And the first commit needs to go

File: sound/soc/sof/intel/hda.h (hunk 1 above)
Comment (fredoh9) [reply sound/soc/sof/intel/hda.h]: sure, will rename it

File: sound/soc/sof/ops.c (hunk 2 above)
Comment (fredoh9) [reply sound/soc/sof/ops.c]: I can change. BTW, I ran checkpatch.pl but there was no warning or error.
is there jenkins's pr_checkpatch.pl available? so I can run before PR.

Comment (ranj063) [reply sound/soc/sof/ops.c]: @fredoh9 did you run it with "--notree --codespell --strict"?

Comment (ranj063) [reply sound/soc/sof/ops.c]: best thing would be to set up the pre and post-commit hooks in your linux repo just like the SOF repo

**Comment by plbossart** [comment]
What @ranj063 said.

File: sound/soc/sof/intel/hda.h (hunk 1 above)
Comment (fredoh9) [reply sound/soc/sof/intel/hda.h]: @ranj063 how about this name? HDA_DSP_REG_POLL_INIT_INTERVAL_US.
A bit long but poll usually pairs with interval.

Comment (ranj063) [reply sound/soc/sof/intel/hda.h]: @fredoh9 yes interval sounds better. you can even drop init and just adjust the interval within the register_poll() method.

Comment (fredoh9) [reply sound/soc/sof/intel/hda.h]: sound great. will upload the update shortly.

**Comment by plbossart** [comment]
I didn't see any comments so missed there was an update. @ranj063, are we good here?

**Comment by ranj063** [comment]
@plbossart looks good to me.

**Pull request merged**